### ISSUE TITLE
The latest master branch cannot connect without a specified database

### ISSUE THREAD

**Issue by pimeys** [blackbeam/mysql_async#81]
Here's a simple test code, using the latest `mysql_async` master:

``` rust
use mysql_async::prelude::*;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
    let pool = mysql_async::Pool::new("mysql://root:prisma@127.0.0.1:3306/");
    let conn = pool.get_conn().await?;
    let result = conn.prep_exec("SELECT 1", ()).await?;

    let (_ /* conn */, num) = result.map_and_drop(|row| {
        let number: i64 = row.get(0).unwrap();
        number
    }).await?;

    println!("We got a result: {:?}", num[0]);
    Ok(())
}
```

This gives a very interesting error:

``` rust
Server(ServerError { code: 1049, message: "Unknown database \'mysql_native_password\'", state: "42000" })
```

We were digging into `mysql_async` and `rust_mysql_common` and noticed [something changed](https://github.com/blackbeam/rust_mysql_common/blob/f72056d9cd6e55e6b07c5e75cdaad972ccf69644/src/packets.rs#L1060) in the protocol recently that might be causing this. The error is gone if we add a database name to the connection string given to the `Pool` initialization.

The sync mysql crate (16.1.0) doesn't have this issue, and the following example works:

``` rust
use mysql as my;

fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
    let pool = my::Pool::new("mysql://root:prisma@127.0.0.1:3306/")?;
    let result = pool.prep_exec("SELECT 1", ())?;

    let num: Vec<i64> = result.map(|row| {
        row.unwrap().get(0).unwrap()
    }).collect();

    println!("We got a result: {:?}", num[0]);
    Ok(())
}
```

**Comment by pimeys** [comment]
Hmm, looks like the sync version uses newer version from the common library. I started refactoring it to the async crate, but I need a bit more context to understand some of the missing functions...

**Comment by blackbeam** [comment]
@pimeys, hi!

Yeah, versions of `mysql_common` prior to 0.19.0 are broken.
I'll patch 0.18 soon. Please use the following workaround until then:

```rust
let mut opts = Opts::from_url(url)?;
if opts.db_name() == Some("") {
    let mut builder = OptsBuilder::from_opts(opts);
    builder.db_name(None::<String>);
    opts = builder.into();
}
```

**Comment by blackbeam** [comment]
Patched version published as `v0.18.1`. Please issue `cargo update`.
